### PR TITLE
Clear fast boot rendered head on browser boot

### PR DESCRIPTION
**Pull request by ronco**
Split initializers into fastboot & browser flavors.

Use placeholder meta tags for marking extent of head.hbs content.

### CODE REVIEW

**Comment by ronco** [comment]
@tomdale @rwjblue comments welcome.

File: app/instance-initializers/browser/head.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Ember from 'ember';
+
+export function initialize(instance) {
+  // clear fast booted head (if any)
+  Ember.$('meta[name="ember-cli-head-start"]')
+    .nextUntil('meta[name="ember-cli-head-end"] ~')
+    .addBack()
+    .remove();
+  const container = instance.lookup ? instance : instance.container;
```
Comment (rwjblue) [inline app/instance-initializers/browser/head.js]: Should be able to remove this conditional, fastboot only supports ember 2.3.0+ so instance.lookup should always be present.

File: app/instance-initializers/browser/head.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  const renderer = container.lookup('renderer:-dom');
+  const component = container.lookup('component:head-layout');
+  component.appendTo(renderer._dom.document.head);
```
Comment (rwjblue) [inline app/instance-initializers/browser/head.js]: In browser mode, can we just always specify document.head?

**Comment by rwjblue** [comment]
This requires a shared package to ensure that fastboot version doesn't run when it ember-cli-fastboot is not present (since this addon still supports that world).

@tomdale this is he scenario that I was explaining on the fastboot/browser initializer stripping PR.

File: app/instance-initializers/browser/head.js (hunk 1 above)
Comment (ronco) [inline app/instance-initializers/browser/head.js]: Good point. :+1:

File: app/instance-initializers/browser/head.js (hunk 2 above)
Comment (ronco) [inline app/instance-initializers/browser/head.js]: I'll give that a try.

**Comment by rwjblue** [comment]
See https://github.com/tildeio/ember-cli-fastboot/pull/110#issuecomment-184680878 for details on filtering the initializers.

**Comment by tomdale** [comment]
Is this good to go?

**Comment by ronco** [comment]
@tomdale haven't tested out the filter module as much as I plan to yet, but I should be able to cut a new release of this addon today.

**Comment by ronco** [comment]
@tomdale just published as 0.0.4

**Comment by tomdale** [comment]
You're the hero Gotham needs.

**Pull request merged**